### PR TITLE
Spike/cross device tests

### PR DESCRIPTION
**Pull request by hjheath**
PR to check multi-tab tests work on CI

There are a few moving parts here:

- Browser
- Web driver
- Selenium client

I had to refactor the textarea element area so that selenium could pull the text out of it.

I have got a basic scenario of the cross device flow working between two tabs. This works locally and on CI. 

As far as I can tell there is nothing specific to chrome. However I have been unable to test this approach on firefox because my firefox webdriver crashes (for unrelated reasons).

I will not merge this PR, but use it as the starting point for writing the cross device tests.

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-262-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://262-pr-onfido-sdk-ui-onfido.surge.sh

File: test/features/step_definitions/sdk_steps.rb [hunk 1]
```diff
@@ -17,3 +18,20 @@
     When I click on #{action_button} ()
   }
 end
+
+element = '(\w+(?:|\(.*\)) \(\w*\)(?:| index \d+))'
```
Comment (dawiddylowicz) [inline test/features/step_definitions/sdk_steps.rb]: Why is this needed?

**Comment by dawiddylowicz** [comment]
Nicely done!

File: test/features/step_definitions/sdk_steps.rb (hunk 1 above)
Comment (hjheath) [reply test/features/step_definitions/sdk_steps.rb]: It's a regex from the monster gem used to identify HTML elements. It's used in line 24.

For the tests I will move the "open new tab" step to the monster gem so we can reuse the regex there.

Comment (dawiddylowicz) [reply test/features/step_definitions/sdk_steps.rb]: Ok I see, thanks

File: src/components/crossDevice/CrossDeviceLink/index.js [hunk 2]
```diff
@@ -217,7 +217,9 @@ class CrossDeviceLinkUI extends Component {
           <div className={style.copyLinkSection}>
             <div className={`${style.label}`}>Copy link instead:</div>
               <div className={classNames(style.actionContainer, {[style.copySuccess]: this.state.copySuccess})}>
-                <textarea className={style.linkText} ref={(textarea) => this.textArea = textarea} value={mobileUrl} />
+                <textarea className={style.linkText} ref={(element) => this.linkText = element}>
```
Comment (rfreitas) [inline src/components/crossDevice/CrossDeviceLink/index.js]: did anything change?

Comment (hjheath) [reply src/components/crossDevice/CrossDeviceLink/index.js]: functionally? I don't think so. But the link text appears in the DOM this way so the tests can parse it

Comment (rfreitas) [reply src/components/crossDevice/CrossDeviceLink/index.js]: saw it now. well, it break the standard I think, because it's a form element. https://stackoverflow.com/questions/5314186/javascript-get-textarea-input-via-value-or-innerhtml

File: test/features/step_definitions/sdk_steps.rb [hunk 3]
```diff
@@ -17,3 +18,20 @@
...
+
+When(/^I open #{element} in a new tab$/) do |element|
+  url = element.text
```
Comment (rfreitas) [inline test/features/step_definitions/sdk_steps.rb]: if the element is textarea the url can be retrieved with `element.value`, that way we respect the standard.

File: src/components/crossDevice/CrossDeviceLink/index.js (hunk 2 above)
Comment (hjheath) [reply src/components/crossDevice/CrossDeviceLink/index.js]: I can revert this - I found a way to pull the value out using selenium:`element.attribute('value')` rather than `element.text`

This element should probably just be a <p></p>, but I don't plan to change it now

Comment (hjheath) [reply src/components/crossDevice/CrossDeviceLink/index.js]: http://www.rubydoc.info/gems/selenium-webdriver/0.0.28/Selenium/WebDriver/Element

File: test/features/step_definitions/sdk_steps.rb (hunk 3 above)
Comment (hjheath) [reply test/features/step_definitions/sdk_steps.rb]: I linked an old version of the docs. In the version we are using `element.value` has been removed so we must use `element.attribute('value')`
